Threaded safe creation

AssafMiron opened the pull request.

infamousjoeg commented:
Will upgrade lab to v11.3 and test out.

AssafMiron commented:
> Will upgrade lab to v11.3 and test out.

Hey @infamousjoeg ,
Any update?

infamousjoeg commented:
> > Will upgrade lab to v11.3 and test out.
> 
> Hey @infamousjoeg ,
> Any update?

Sorry for the delay, testing this now...

infamousjoeg commented on Safe Management/Invoke-SafeManagement.ps1:
```diff
@@ -0,0 +1,914 @@
+				}
+				else
+				{
+					Get-Safes
```
```
PS /Users/joegarcia/go/src/github.com/cyberark/epv-api-scripts/Safe Management> ./Invoke-SafeManagement.ps1 -PVWAURL https://cyberark.joegarcia.dev/PasswordVault -List                                                                     Script Started                                                                                                                                                                                                                              Setting script to use TLS 1.2                                                                                                                                                                                                                                                                                                                                                                                                                                                           Safe Management
Enter your User name and Password
User: jgarcia
Password for user jgarcia: ************

Retrieving Safes...
Retrieving safes from the vault...
There was an error retrieving the safes from the Vault. The error was:
Write-Error: /Users/joegarcia/go/src/github.com/cyberark/epv-api-scripts/Safe Management/Invoke-SafeManagement.ps1:752
Line |
 752 |                      Get-Safes
     |                      ~~~~~~~~~
     | Exception of type 'Microsoft.PowerShell.Commands.WriteErrorException' was thrown.

Logoff Session...
Script ended
```

infamousjoeg commented on Safe Management/Invoke-SafeManagement.ps1:
```diff
@@ -0,0 +1,914 @@
+					{
+						# Create one Safe
+						Write-Host "Adding the safe $SafeName..." -ForegroundColor Yellow
+						Create-Safe @parameters
```
```
PS /Users/joegarcia/go/src/github.com/cyberark/epv-api-scripts/Safe Management> ./Invoke-SafeManagement.ps1 -PVWAURL https://cyberark.joegarcia.dev/PasswordVault -Add
Script Started                                                                                                                                                                                                                              Setting script to use TLS 1.2                                                                                                                                                                                                                                                                                                                                                                                                                                                           Safe Management
Enter your User name and Password
User: jgarcia
Password for user jgarcia: ************

Adding the safe ...

cmdlet Create-Safe at command pipeline position 1
Supply values for the following parameters:
safename: Test-Safe123
Adding the safe Test-Safe123 to the Vault...
Error adding Test-Safe123 to the Vault. The error was:
Write-Error: /Users/joegarcia/go/src/github.com/cyberark/epv-api-scripts/Safe Management/Invoke-SafeManagement.ps1:809
Line |
 809 |                          Create-Safe @parameters
     |                          ~~~~~~~~~~~~~~~~~~~~~~~
     | Exception of type 'Microsoft.PowerShell.Commands.WriteErrorException' was thrown.

Logoff Session...
Script ended
```

infamousjoeg commented:
@AssafMiron I'm testing on PowerShell Core and not experiencing behavior I would expect out of this script.  What PS Version did you happen to test this on yourself during development?

AssafMiron commented:
@infamousjoeg I'm using Powershell 5 for this
From the errors you encountered these are even without the Jobs and threaded mode so I guess the error is around the Logon Header
I changed it recently from Global to non-global and did not have time to check it yet, that might caused the issues
I added a way now to send the Logon header to all methods so that might solve things here
try running the script with -Debug -Verbose, this should show the REST Error and then we can verify it

Thanks,
Assaf

bab29 commented:
@AssafMiron 

This is almost a year old, do we need it or can I go ahead and close it?

AssafMiron commented:
Think this can be closed
I think there is also anPR waiting for it, not sure if it was fully tested
or not

On Thu, Dec 16, 2021, 18:36 bab29 ***@***.***> wrote:

> @AssafMiron <https://github.com/AssafMiron>
>
> This is almost a year old, do we need it or can I go ahead and close it?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cyberark/epv-api-scripts/pull/24#issuecomment-995983931>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC3PN3MKHI5GFG66JHII23TURIIPVANCNFSM4LTPS65Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

bab29 commented:
I will close this and if needed we can create a new one.
